### PR TITLE
A design for timing of storage proofs

### PR DESCRIPTION
**Pull request by markspanbroek**
Added a document that describes the current design of the proof timing in [our smart contracts](https://github.com/status-im/dagger-contracts), the problems that we encounter, and a new design that intends to solve these problems.

### CODE REVIEW

File: design/storage-proof-timing.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+     m           255                      ^           0
+     e                                    |
+     |                                  pointer
+     |
+     |      - - - |-----------------------------------|
+     |           255                  ^               0
+     |                                |
+     v                             pointer
+
+Pointer duos
```
Comment (dryajov) [inline design/storage-proof-timing.md:123]: What if we simply use a smaller modulo, wouldn't that give us a window that will keep the pointer valid for that many blocks - `WINDOW = 256 - (blocknumber + period) % modulo`? For example, if the modulo is 128, then this gives the node a `WINDOW` of at least 128 blocks where the pointer is still valid. Once the `WINDOW` has expired the proof is impossible to verify? We can adjust the `WINDOW` depending on the average block production cadence or some other parameter, even wall clock?

Comment (dryajov) [reply design/storage-proof-timing.md:123]: Obviously we adjust the window by tweaking the `modulo`.

**Comment by markspanbroek** [comment]
Replaced the pointer duos by a strategy in which the pointer is ignored some of the time, as suggested by @cskiraly.

**Comment by Menduist** [comment]
If you ever feel bad about the wasted time, remember the VGA blanking time:
https://i.stack.imgur.com/6wpSA.jpg

**Pull request merged**